### PR TITLE
Documentation check in CI/CD Pipeline

### PR DESCRIPTION
**Pull request by Durganshu**
In this PR, an automated test routine is being introduced to check the completeness of the documentation while running the pipeline.

If any of the header files have missing documentation (except files in parsers/, external_interfaces/, examples/, data-sources/ and .cpp files), the pipeline fails.

### CODE REVIEW

File: src/Component.hpp [hunk 1]
```diff
@@ -634,6 +634,11 @@ class Node : public Component {
     ~Node() override = default;
 #ifdef PROC_CPUINFO
 public:
+    /**
+     * @private
```
Comment (stepanvanecek) [inline src/Component.hpp:638]: does this @private mean it is not included in the API documentation? if yes, this call should be public

File: src/Component.hpp [hunk 2]
```diff
@@ -699,6 +704,10 @@ class Memory : public Component {
 
 #ifdef NVIDIA_MIG
 public:
+    /**
+     * @private
```
Comment (stepanvanecek) [inline src/Component.hpp:708]: same as above

File: src/Component.hpp [hunk 3]
```diff
@@ -963,6 +972,10 @@ class Cache : public Component {
...
```
Comment (stepanvanecek) [inline src/Component.hpp:976]: same as above

File: src/Component.hpp [hunk 4]
```diff
@@ -1093,8 +1106,22 @@ class Core : public Component {
 
 #ifdef PROC_CPUINFO
 public:
+    /**
+    * @private
```
Comment (stepanvanecek) [inline src/Component.hpp:1110]: same as above

File: src/Component.hpp [hunk 5]
```diff
@@ -1093,8 +1106,22 @@ class Core : public Component {
...
+    * Refreshes the frequency of the core.
+    */
     int RefreshFreq(bool keep_history = false);
+
+    /**
+    * @private
```
Comment (stepanvanecek) [inline src/Component.hpp:1116]: same as above

File: src/Component.hpp [hunk 6]
```diff
@@ -1093,8 +1106,22 @@ class Core : public Component {
...
+    * Sets the frequency of the core.
+    */
     void SetFreq(double _freq);
+
+    /**
+    * @private
```
Comment (stepanvanecek) [inline src/Component.hpp:1122]: same as above

File: src/Component.hpp [hunk 7]
```diff
@@ -1130,7 +1157,16 @@ class Thread : public Component {
 
 #ifdef PROC_CPUINFO //defined in proc_cpuinfo.cpp
 public:
+    /**
+    * @private
```
Comment (stepanvanecek) [inline src/Component.hpp:1161]: same as above

File: src/Component.hpp [hunk 8]
```diff
@@ -1130,7 +1157,16 @@ class Thread : public Component {
...
+    * Refreshes the frequency of the thread.
+    */
     int RefreshFreq(bool keep_history = false);
+
+    /**
+    * @private
```
Comment (stepanvanecek) [inline src/Component.hpp:1167]: same as above

File: src/xml_dump.hpp [hunk 9]
```diff
@@ -6,8 +6,22 @@
 #include "Component.hpp"
 #include "DataPath.hpp"
 
+/**
+ * @private
```
Comment (stepanvanecek) [inline src/xml_dump.hpp:10]: same as above

File: src/Component.hpp (hunk 1 above)
Comment (Durganshu) [reply src/Component.hpp:638]: Yes, fixed!

**Pull request merged**